File: samples/VBHelloWorld/VBGrains/HelloGrain.vb
Imports Orleans

''' <summary>
''' Orleans grain implementation class Grain1
''' </summary>
Public Class HelloGrain
    Inherits Grain
    Implements Interfaces.IHelloGrain

    Public Function SayHello(greeting As String) As Task(Of String) Implements Interfaces.IHelloGrain.SayHello
        Return Task.FromResult("Hello, this is Visual Basic speaking!")
    End Function

End Class

File: samples/Deployment/AzureContainerApps/Grains/HelloGrain.cs
using Abstractions;
using Orleans;

namespace Grains
{
    public class HelloGrain : Grain, IHelloGrain
    {
        public Task<string> SayHello()
        {
            return Task.FromResult($"Hello from Grain {this.GetPrimaryKeyString()}");
        }
    }
}